pkg/app/api/grpcapi/web_api.go

```diff
@@ -200,15 +205,20 @@ func (a *WebAPI) DeleteEnvironment(ctx context.Context, req *webservice.DeleteEn
 	// Check if no Piped has permission to the given environment.
 	pipeds, err := a.pipedStore.ListPipeds(ctx, datastore.ListOptions{
 		Filters: []datastore.ListFilter{
+			{
+				Field:    "ProjectId",
+				Operator: datastore.OperatorEqual,
+				Value:    claims.Role.ProjectId,
+			},
 			{
 				Field:    "EnvIds",
 				Operator: datastore.OperatorContains,
 				Value:    req.EnvironmentId,
 			},
 			{
-				Field:    "ProjectId",
+				Field:    "Disabled",
 				Operator: datastore.OperatorEqual,
-				Value:    claims.Role.ProjectId,
+				Value:    false,
 			},
```

Comment:
In case we do not (or maybe can't) index on the array field, this `Disabled` field should be arranged before the `EnvIds` field. 👀

Reply:
We don't know exactly what is going; I am looking to see how it goes at our dev environment with this order once got merged.

Reply:
Got it 🙆‍♀️

Reply:
🤒